packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -1747,6 +1747,44 @@ void main() {
     },
   );
 
+  testWidgets(
```

Comment:
Nit: Can any of these tests that are more or less duplicated between CupertinoTextField, TextField, etc. be moved to only EditableText's tests to avoid the duplication?

Reply:
The reason I kept the multiple versions was because each of them has the duplicated logic, so it needs to be tested in each place, in case only one of the sites gets modified.

Reply:
Ah of course, makes sense to keep that if my idea above doesn't work out.